versions/development/SPEC.md

```diff
@@ -358,6 +359,65 @@ Strings can also contain the following types of escape sequences:
 * A hexadecimal escape code starts with `\x`, followed by 2 hexadecimal digits `0-9a-fA-F`. 
 * A unicode code point starts with `\u` followed by 4 hexadecimal characters or `\U` followed by 8 hexadecimal characters `0-9a-fA-F`.
 
+##### Multi-line Strings
+
+Strings that begin and end with three consecutive single- or double-quotes may span multiple lines. The opening quotes of a multi-line string may be followed by whitespace (optional) and must have a newline before the first non-whitespace character; these leading whitespace/newline characters are removed. All subsequent non-empty lines are then used to determine the multi-line string's *common leading whitespace* - the minimum number of whitespace characters occuring before the first non-whitespace character in a line or the end of the line, whichever comes first. This common leading whitespace is stripped from the beginning of all the lines in the multi-line string.
+
+```wdl
+# These strings are equivalent. The middle lines strings B, C, and D are empty and so do not count
+# towards the common leading whitespace determination.
+
+String multi_line_A = """
+    this is a
+  a multi-line string"""
+
+String multi_line_B = '''
+        this is a
+
+          multi-line string'''
+
+String multi_line_C = """
+  this is a
+
+    multi-line string"""
+
+String multi_line_D = """
+this is a
+
+  multi-line string"""
+```
+
+Newline characters are not stripped out unless they are escaped, i.e. when the last character of a line is `\`.
+
+```wdl
+# These two strings are equivalent:
+String single_line = "this is a double-quoted string that contains no newlines"
+String multi_line_escaped = """
+  this is a \
+  double-quoted string \
+  that contains no newlines"""
+```
+
+Keep in mind that if the ending quotes are on a line by themselves and are preceeded by whitespace, that line is included when determining the common leading whitespace.
+
+```wdl
+# The following two strings are equivalent. Even though the first line of `s2` is indented by six 
+# spaces, the common leading whitespace in this string is 2, due to the two spaces preceeding the 
+# closing quotes.
+String s1 = "    text indented by 4 spaces"
+String s2 = """
+      text indented by 4 spaces
+  """
```

Comment:
Notwithstanding the above comments on this line, I'm also not sure why wouldn't s2 end with a newline? (i.e. shouldn't s1 end in a newline to equal s2?)

Reply:
good catch, fixed